src/pybind/mgr/dashboard/tests/test_grafana.py

```diff
@@ -41,23 +42,32 @@ def test_url(self):
         self.assertStatus(200)
         self.assertJsonBody({'instance': 'http://localhost:3000'})
 
-    def test_validation(self):
+    @patch('dashboard.controllers.grafana.GrafanaRestClient.url_validation')
+    def test_validation_endpoint_returns(self, url_validation):
+        """
+        The point of this test is to see that `validation` is an active endpoint that returns a 200
+        status code.
+        """
+        url_validation.return_value = b'404'
         self.server_settings()
         self._get('/api/grafana/validation/foo')
-        self.assertStatus(500)
+        self.assertStatus(200)
+        self.assertBody(b'"404"')
 
     def test_dashboards_unavailable_no_url(self):
-        self.server_settings(url=None)
+        self.server_settings(url="")
         self._post('/api/grafana/dashboards')
         self.assertStatus(500)
 
-    def test_dashboards_unavailable_no_user(self):
-        self.server_settings(user=None)
+    @patch('dashboard.controllers.grafana.GrafanaRestClient.push_dashboard')
+    def test_dashboards_unavailable_no_user(self, pd):
+        pd.side_effect = DashboardException
```

Comment:
`push_dashboard` method doesn't raise `DashboardException`.
(Not affect the test logic)

Reply:
All true. When I quickly thought about it, it was better than using a Python built-in exception. But now I think it'd even be better if it was a requests exception. But yes, it doesn't matter for the logic which exception is raised, I just was unhappy with any Python-native exception.

As I'll have to restart make-check anyway, what do you think about using a `RequestException` instead?